utils/preferences/src/extensions/unicode/keywords/collation.rs

```diff
@@ -0,0 +1,11 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+crate::enum_keyword!(Collation {
+    "standard" => Standard,
+    "search" => Search,
+    "phonetic" => Phonetic,
+    "pinyin" => Pinyin,
+    "searchjl" => Searchjl
+}, "co");
```

Comment:
I think "search" doesn't belong here, since it's for a different mode of usage. It makes sense for the user to have a preference between "stroke" and "zhuyin". It does not make sense for the user to express a preference between "stroke" and "search".

I'm not quite sure what the exact semantics of "search" vs. "searchjl" are, but 1) we don't have a search API at this point anyway and 2) I believe it's a contextual application design decision whether a given search operation with the Korean UI locale should use "search" or "searchjl".

So let's not put either "search" or "searchjl" here.

That isn't enough to decide what to put here.

As I understand it, the main use case for this is deciding whether for Traditional Chinese you want `zhuyin` instead of the default-for-`zh-Hant` `stroke`. `stroke` makes sense for any of the varieties of Chinese (Mandarin, Cantonese, etc.), but `zhuyin` is Mandarin-focused. (I don't know if using `zhuyin` sorting for e.g. Hakka is nonetheless a thing that user do.)

From theorizing in the reverse, one might except it to make sense for a user to override the default Mandarin-focused `pinyin` collation order of `zh-Hans` to `stroke` when not using Mandarin, but I don't know what the actual user practice is.

Other than that choosing between Mandarin-phonetic (`pinyin` and `zhuyin`) vs. `stroke` for `zh`, it's not really clear how strong the use cases are. `compat` for Arabic looks a lot like something that's about ICU change management from long ago (as were `big5han` and `gb2312`). Is `standard` vs. `dict` for Sinhala something that makes sense as a user preference? Is `standard` vs. `phonebk` for German something that makes sense as a user preference? Theoretically, `standard` vs. `trad` for Spanish looks potentially something that could make sense as user preference, but what's the level of demand, really? `standard` vs. `trad` for Finnish and Swedish is being removed. I have no idea of the user demand for `standard` vs. `trad` for Bangla.

Reply:
perfect, I appreciate your feedback and I'm glad to discuss it in context of introducing particular keys. For now I wanted to verify that the architecture works for us. We'll discuss this key in a PR for "co".